crates/matrix-sdk-ui/src/timeline/inner/state.rs

```diff
@@ -682,7 +682,7 @@ impl TimelineInnerStateTransaction<'_> {
         self.read_receipts.clear();
         self.reactions.clear();
         self.fully_read_event = None;
-        self.event_should_update_fully_read_marker = false;
+        self.has_up_to_date_read_marker_item = false;
```

Comment:
We changed the default here from `should_update = false` to `has_up_to_date = false` which looks like a change in meaning. Was that deliberate?

Reply:
Yes. In this case, it was even probably a bug: if you received a limited sync (that clears the timeline, right now), then the read marker would just disappear from existence, even if we could place it correctly, until a new one was received from the sync. Now we retry to put the read marker the first time an event will be added after a limited sync, which is less surprising in terms of UX in my opinion.

Reply:
Great! If you think a comment here explaining the above would help, I'd support that.